### PR TITLE
Improvements to CLI

### LINKED ISSUE #23: Clarify auto-caching in `jcache execute`
From #16 

Successfully executed notebooks are automatically cached. Message(s) in `jcache execute` should be added make this clear.

### LINKED ISSUE #24: Rename `add-many` / `add-one`
> I still feel a clash about `add-many` vs. `add-one`. What if add-many became just `add`, and `add-one` became `add-with-artefacts`?

Yep that can be done. Similarly this could be done for staging `add-one` -> `add-with-assets`

_Originally posted by @chrisjsewell in https://github.com/ExecutableBookProject/jupyter-cache/issues/16#issuecomment-594958966_

### LINKED ISSUE #26: Improve `KeyError` message
> If I try to merge in to a notebook that's not in the cache, I get KeyError: <hash>. This should probably be a more user-friendly error like "The notebook you gave doesn't exist in the cache"

👌 

_Originally posted by @chrisjsewell in https://github.com/ExecutableBookProject/jupyter-cache/issues/16#issuecomment-596263654_

### LINKED ISSUE #32: Discuss how to visualise historical cache records
In #16 I mentioned that it was a bit dis-orienting to see multiple versions of the same notebook in the cache list. This is an issue to discuss how that should be handled.

One option would be to include some kind of "Origin URI" value, to make it clear that the cache entries are pointing to the same file.

I'd actually lean towards trying to find a way to *only* show a single entry for a given notebook with the default `list`. It seems that the use-case where somebody wants to list the historical cache records of a notebook to grab one from history from the CLI would be relatively rare. To avoid this confusion, perhaps `jcache cache list` could have a flag like `--include-history` that would include historical cached items, and if not given, the cache would only show the *latest* version of each item.

### PR DESCRIPTION
**Pull request by chrisjsewell**
See commit messages and README

fixes #23, fixes #24, fixes #26, fixes #32

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,19 +139,21 @@ You can remove cached records by their ID.
...
-   4  tests/notebooks/complex_outputs.ipynb  2020-02-29 17:48  2020-02-29 17:48
-   3  tests/notebooks/basic_unrun.ipynb      2020-02-29 17:48  2020-02-29 17:48
-   2  tests/notebooks/basic_failing.ipynb    2020-02-29 17:48  2020-02-29 17:48
+   5  tests/notebooks/external_output.ipynb  2020-03-12 17:31  2020-03-12 17:31
+   4  tests/notebooks/complex_outputs.ipynb  2020-03-12 17:31  2020-03-12 17:31
+   3  tests/notebooks/basic_unrun.ipynb      2020-03-12 17:31  2020-03-12 17:31
+   2  tests/notebooks/basic_failing.ipynb    2020-03-12 17:31  2020-03-12 17:31
 ```
 
+Tip: Use the `--latest-only` option, to only show the latest versions of cached notebooks.
```
Comment (choldgraf) [inline README.md]: What do you think about the default being `--latest-only` and having a flag like `--include-history` instead?

Comment (chrisjsewell) [reply README.md]: The reason why I'm hesitant to use `--include-history` is that the terminology is not strictly *true* in how the cache actually operates. Filtering for a certain `Origin URI` and sorting by `Created` **does not** guarantee to show a history for that URI; for example if you start changing file names of notebooks, or for a specific example:

- create a git repository with `notebook.ipynb` and cache it
- switch the branch of the git repository, modify `notebook.ipynb` and cache that
- switch back to the original git branch

Assuming the actual cache isn't committed to git, you now have two versions of `notebook.ipynb` in the cache and the **older** one is actually the latest for the branch you are currently on.

Comment (chrisjsewell) [reply README.md]: #34 could be an alternative way to show only particular records

Comment (choldgraf) [reply README.md]: Yeah, that makes sense. I am just trying to think of the most common use-case, and generally speaking we should be tailoring the default UI around the common use-cases. I think it's much more common for someone to think "I want to see a list of the notebooks that are in the cache" than for them to also think "and I want to see  values for every version that I have ever cached for a given notebook".

Though your branching example is a tricky one, that would definitely confuse people :-/

Comment (chrisjsewell) [reply README.md]: I’m definitely open to some form of solution, but it should be carefully considered what the balance is between ease-of-use and compliance with underlying logic

**Pull request merged**